proposals/20200504-falco-artifacts-scope.md

```diff
@@ -0,0 +1,191 @@
+# Falco Scope
+
+As a project we would like to support the following artifacts.
+
+Everything else will be moved to [contrib](https://github.com/falcosecurity/contrib).
+
+As a project we will build, change, rename, and move files, documents, scripts, configurations according to this new state of the world.
+
+Inspired from [#1114](https://github.com/falcosecurity/falco/issues/1114) and many of the weekly community calls.
+
+## Terms
+
+**falco** 
+
+*Standalone Falco binary*
+
+**driver**
+
+*System call provider from the Linux kernel. Either (`bpf`, `module`, or `ptrace`)*
+
+**falco-driver-loader-script**
+
+
+*The bash script found [here](https://github.com/falcosecurity/falco/blob/master/scripts/falco-driver-loader) that tries to compile else download the kernel module.*
+
+**package**
+
+*An installable artifact that is operating system specific. All packages MUST be hosted on bintray for X86_64.*
+
+**image**
+
+*OCI compliant container image hosted on dockerhub with tags for every release.*
+ 
+
+# Packages
+
+Official packages for X86_64 only. All package names should contain a version which is not specified below.
+
+The following convention MUST be used for all packages.
+
+_If a package installs Falco as a daemon (systemd, init) it MUST contain `daemon`._
+
+_If a package installs the Falco kernel module it MUST contain `module`._
+
+_If a package installs the Falco BPF probe it MUST contain `bpf`._
+
+_If a package installs the Falco kernel module it MUST contain `module`._
+
+---
+
+### .deb
+
+ Falco running in debian like systems that will default to the kernel module.
+
+- falco-x86.deb
+     - `falco` only (default depends on `falco-module`)
+ - falco-daemon-x86.deb
+     - `falco` and script to configure as daemon (default depends on `falco-module`)
+ - falco-module-x86.deb
+     - `module` only
+ - falco-bpf-x86.deb
+     - `bpf` only
+ - falco-ptrace-x86.deb
+     - `ptrace` only
+
+### .rpm
+
+ Falco running in rpm like systems that will default to the kernel module.
+
+- falco-x86.deb
+     - `falco` only (default depends on `falco-module`)
+ - falco-daemon-x86.deb
+     - `falco` and script to configure as daemon (default depends on `falco-module`)
+ - falco-module-x86.deb
+     - `module` only
+ - falco-bpf-x86.deb
+     - `bpf` only
+ - falco-ptrace-x86.deb
+     - `ptrace` only
+
+
+### .tar.gz
+
+ - falco-src-x86.tar.gz
+     - No binaries
+     - `INSTALL` file
+ - falco-driver-src-x86.tar.gz
+     - `falco` and `/driver` with `Makefile`
+     - `INSTALL` file
+ - falco-driver-download-x86.tar.gz
+     - `falco` and `falco-driver-loader` download only
+ - falco-driver-full-x86.tar.gz
+     - `falco` and `falco-driver-loader` download else compile
+
+
+# Images
```

Comment:
I think that during the discussion (and followup reviews, and implementations) of dockerfiles we decided to remove some of them and keep only those identifying well specific use-cases (e.g., https://github.com/falcosecurity/falco/issues/1019). 

And we decided to stick with the names we already had in the majority of cases (except the rename of `probeloader` into `driverloader` to better stick to your branding proposal).

Changing names in this way would confuse even me that work daily on the project, honestly.

Furthermore, I believe that naming things after (part of) their behavior is an anti-pattern.
To tool to do that is documentation IMHO, not naming. Even more, because complex behavior and/or implementations cannot be expressed with a single name (even tho it is composed of some words).

Thus, I think the issue here is better documenting the docker image Falco has, not renaming them or change the behavior they have while keeping an existing name.

Reply:
Apart from revamping the documentation on falco.org (which is very needed),

an approach to document the usage of the images could be the one I used for `falco-builder`, source [here](https://github.com/falcosecurity/falco/blob/master/docker/builder/root/usr/bin/usage).

In this way, a user can run the `falco-builder` with the `usage` command and obtain all the information needed to use it effectively.

We could adopt the same approach also for other docker images.

Reply:
We decided to provide an `usage` command for every official docker image we have.

We updated the proposals accordingly. Resolving this :)